apps/webapp/src/app/(Magic-Link)/magic-link/page.tsx

```
@@ -0,0 +1,614 @@
+
+  const filteredProviders = data.filter(
+    (provider) =>
+      provider.name.toLowerCase().includes(searchTerm.toLowerCase()) &&
+      (selectedCategory === "All" ||
+        provider.vertical?.toLowerCase() === selectedCategory.toLowerCase())
+  );
+
+  const handleProviderSelect = (provider: Provider) => {
+    setSelectedProvider({
+      provider: provider.name.toLowerCase(),
+      category: provider.vertical!.toLowerCase(),
```

Biome

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)